apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/(withTabs)/traces/_components/Trace.tsx

```diff
@@ -0,0 +1,164 @@
+import { SplitPane } from '@latitude-data/web-ui/atoms/SplitPane'
+import { Text } from '@latitude-data/web-ui/atoms/Text'
+import { useCallback, useEffect, useRef, useState } from 'react'
+import { TimelineGraph } from '$/components/tracing/traces/Timeline/Graph'
+import { TimelineScale } from '$/components/tracing/traces/Timeline/Scale'
+import { TimelineTree } from '$/components/tracing/traces/Timeline/Tree'
+import { useTrace } from '$/stores/traces'
+import { Icon } from '@latitude-data/web-ui/atoms/Icons'
+import { useSelectedSpan } from './SelectedSpansContext'
+import { AssembledSpan, AssembledTrace } from '@latitude-data/core/constants'
+
+export function Trace({ traceId }: { traceId: string }) {
+  // For now, just show basic span information
+  // This should be expanded to show the full trace timeline
+  const { data: trace, isLoading } = useTrace({ traceId })
+
+  if (isLoading) {
+    return (
+      <div className='w-full h-full flex items-center justify-center gap-2 p-4 bg-secondary'>
+        <Icon name='loader' color='foregroundMuted' className='animate-spin' />
+        <Text.H5 color='foregroundMuted'>Assembling trace</Text.H5>
+      </div>
+    )
+  }
+
+  if (!trace || trace.children.length < 1) {
+    return (
+      <div className='w-full h-full flex items-center justify-center gap-2 p-4 bg-secondary'>
+        <Text.H5 color='foregroundMuted'>No events found so far</Text.H5>
+      </div>
+    )
+  }
+
+  return (
+    <div className='w-full h-full max-h-96 overflow-y-auto custom-scrollbar'>
+      <Timeline trace={trace} />
+    </div>
+  )
+}
+
+const TREE_MIN_WIDTH = 125 // 125px
+const GRAPH_MIN_WIDTH = 450 // 450px
+
+function findSpanByIdBFS(
+  trace: AssembledTrace,
+  spanId?: string | null,
+): AssembledSpan | null {
+  const queue: AssembledSpan[] = [...trace.children]
+  if (!spanId) return null
+
+  while (queue.length > 0) {
+    const currentSpan = queue.shift()!
+
+    if (currentSpan.id === spanId) {
+      return currentSpan
+    }
+
+    if (currentSpan.children && currentSpan.children.length > 0) {
+      queue.push(...currentSpan.children)
+    }
+  }
+
+  return null
+}
+
+function Timeline({ trace }: { trace: AssembledTrace }) {
```

Comment:
again, mostly reimplemented from existing component, will remove the old one once i'm done